packages/app/src/cli/models/extensions/specifications/transform/app_config_webhook.ts

```diff
@@ -24,12 +24,96 @@ export function transformToWebhookConfig(content: object) {
   const serverWebhooks = getPathValue(content, 'subscriptions') as NormalizedWebhookSubscription[]
   if (!serverWebhooks) return webhooks
 
-  const webhooksSubscriptions: WebhooksConfig['subscriptions'] = []
+  const subscriptions = serverWebhooks.map((subscription) => {
+    const {topic, ...otherFields} = subscription
+    return {...otherFields, topics: [subscription.topic]} as WebhookSubscription
+  })
+  const webhooksSubscriptions: WebhooksConfig['subscriptions'] = mergeAllWebhooks(subscriptions)
 
-  for (const {topic, ...otherFields} of serverWebhooks) {
-    webhooksSubscriptions.push({topics: [topic], ...otherFields})
-  }
-
-  const webhooksSubscriptionsObject = webhooksSubscriptions.length > 0 ? {subscriptions: webhooksSubscriptions} : {}
+  const webhooksSubscriptionsObject = webhooksSubscriptions ? {subscriptions: webhooksSubscriptions} : {}
   return deepMergeObjects(webhooks, {webhooks: webhooksSubscriptionsObject})
 }
+
+export function mergeAllWebhooks(subscriptions: WebhookSubscription[]): WebhookSubscription[] | undefined {
+  if (subscriptions.length === 0) return
+  const topicSubscriptions = subscriptions
+    .filter((subscription) => subscription.topics !== undefined)
+    .map((subscription) => {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      const {compliance_topics, ...rest} = subscription
+      return {topics: subscription.topics, ...rest}
+    })
+  const complianceSubscriptions = subscriptions
+    .filter((subscription) => subscription.topics === undefined || subscription.compliance_topics !== undefined)
```

Comment:
why do we want to check `subscription.topics === undefined`? is checking the `compliance_topics` not sufficient?

Reply:
this makes sure that we're parsing `compliance_topics` from the `privacy_compliance_subsriptions` spec rather than the `webhooks` spec! so for example when we have the TOML like this:

```
[[webhooks.subscriptions]]
topics = ["products/create"]
compliance_topics = ["customers/redact"]
uri = "https://example.com/webhooks"
```

we won't run into the bug where we have 2 compliance subscriptions